### PR TITLE
Add original resolver name in migrated Secret Scanning alerts 

### PR DESCRIPTION
**Pull request by theztefan**
This PR enhances the migration of Secret Scanning alerts (`gh gei migrate-secret-alerts` command) by adding support for the `ResolverName` property and migrates it in the comment of a possible matching target alert. With this we maintain the information of who originally resolved an alert.

Basically addresses #1334.

### Changes Made:

- Updated the `GithubSecretScanningAlert` model to include a new `ResolverName` field.
- Modified the `GithubApi` class to populate `ResolverName` from the GitHub API response.
- Enhanced the resolution comment format in `SecretScanningAlertService` to include the `ResolverName` in the targetResolutionComment.
- Added new test to cover edge-cases and update existing ones.

This improvement provides better visibility into the resolution history of Secret Scanning alerts by including the name of the resolver when applicable.

---

- [x] Did you write/update appropriate tests  
- [x] Release notes updated (if appropriate)  
- [x] Appropriate logging output  
- [x] Issue linked  
- [ ] ~~Docs updated (or issue created)~~  
- [ ] ~~New package licenses are added to `ThirdPartyNotices.txt` (if applicable)~~

### CODE REVIEW

File: src/Octoshift/Services/SecretScanningAlertService.cs [hunk 1]
```diff
@@ -74,9 +74,11 @@ public virtual async Task MigrateSecretScanningAlerts(string sourceOrg, string s
 
                         _log.LogInformation($"  updating target alert:{targetAlert.Alert.Number} to state:{sourceAlert.Alert.State} and resolution:{sourceAlert.Alert.Resolution}");
 
+                        var targetResolutionComment = $"[@{sourceAlert.Alert.ResolverName}] {sourceAlert.Alert.ResolutionComment}";
```
[the remark this answers is not in the document]
Comment (theztefan) [reply src/Octoshift/Services/SecretScanningAlertService.cs:77]: No need as we only "migrate" resolve alerts which must have at least a resolver (and optionally a comment)

File: src/OctoshiftCLI.Tests/Octoshift/Services/SecretScanningAlertServiceTests.cs [hunk 2]
```diff
@@ -740,7 +754,7 @@ public async Task Migrate_Matching_Alerts_With_Different_Resolutions()
             100,
             SecretScanningAlert.AlertStateResolved,
             SecretScanningAlert.ResolutionRevoked,
-            null)
+            "[@] ")
```
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/Octoshift/Services/SecretScanningAlertServiceTests.cs:757]: Since this is a situation which should never happen it'd be better to add the resolver name to the target secret's payload.

**Review by ArinGhazarian — CHANGES_REQUESTED** [review]
Looks good just two minor comments.

File: src/OctoshiftCLI.Tests/Octoshift/Services/SecretScanningAlertServiceTests.cs (hunk 2 above)
Comment (theztefan) [reply src/OctoshiftCLI.Tests/Octoshift/Services/SecretScanningAlertServiceTests.cs]: I've improved the test to make it a bit more clear as I wasn't sure if you're comment was towards the test or the overall unexpected case of source and target alert having different resolution reasons.

If it's the later, agree that it shouldn't happen, but for consistency I suggest we keep the original alerts resolution reason, comment and username as latest update on the target "migrated" alert.

File: src/Octoshift/Services/GithubApi.cs [hunk 3]
```diff
@@ -1182,6 +1182,9 @@ private static GithubSecretScanningAlert BuildSecretScanningAlert(JToken secretA
             ResolutionComment = (string)secretAlert["resolution_comment"],
             SecretType = (string)secretAlert["secret_type"],
             Secret = (string)secretAlert["secret"],
+            ResolverName = secretAlert["resolved_by"]?.Type != JTokenType.Null
+                ? (string)secretAlert["resolved_by"]["login"]
+                : null
```
[the remark this answers is not in the document]
Comment (ArinGhazarian) [reply src/Octoshift/Services/GithubApi.cs]: Yes, your are right and it was my bad! The returned value for `null`s is a `JValue` instead of a `null`, my understanding was that we won't even have the `resolved_by` field if it's `null`! I think your original solution was better.

File: src/OctoshiftCLI.Tests/Octoshift/Services/SecretScanningAlertServiceTests.cs (hunk 2 above)
Comment (ArinGhazarian) [reply src/OctoshiftCLI.Tests/Octoshift/Services/SecretScanningAlertServiceTests.cs]: Yeah my comment was for the later, but I am 👍 if you think it's better to keep it's better as is. Probably I should've marked this as a nitpick.

**Review by ArinGhazarian — COMMENTED** [review]
The code looks good now. I am 👍 once you apply your original code for `secretAlert["resolved_by"]`

**Comment by theztefan** [comment]
@ArinGhazarian pushed [8542c03](https://github.com/github/gh-gei/pull/1337/commits/8542c0394a7f91dc992a72bee492984bc5d5b9e7). As per the conversations above, I think this is not 🚀 :)

**Review by begonaguereca — APPROVED** [review]
This looks great! The implementation is clean, the additional ResolverName handling is well-structured, and the test coverage is excellent—including thoughtful edge cases like null ResolutionComment.

Nice contribution!
